modules/data.remote/inst/s2lai.py

```diff
@@ -0,0 +1,80 @@
+#!/usr/bin/env python3
+# -*- coding: utf-8 -*-
+
+"""
+Calculates LAI from Sentinel 2 data. Uses geeapi to download data and the python implementation of the SNAP toolbox to calculate LAI.
+
+Requires Python3
+
+Warning: Point coordinates as input has currently not been implemented.
+
+Author: Ayush Prasad
+"""
+
+import satellitetools.biophys_xarray as bio
+from satellitetools import gee
+import geopandas as gpd
+import os
+
+
+def s2_lai(geofile, outdir, start, end, qi_threshold):
+    """ 
+    Downloads sentinel 2 data, calculates LAI and saves it in a netCDF file at the specified location.
+    
+    Parameters
+    ----------
+    
+    geofile (str) -- path to the file containing the coordinates and name of the AOI, currently tested with geojson. 
+    
+    outdir (str) -- path to the directory where the output file is stored. If specified directory does not exists, it is created.
+  
+    start (str) -- starting date of the data request in the form YYYY-MM-DD
+    
+    end (str) -- ending date of the data request in the form YYYY-MM-DD
+    
+    qi_threshold (float) -- From satellitetools: Threshold value to filter images based on used qi filter. qi filter holds labels of classes whose percentages within the AOI is summed. If the sum is larger then the qi_threshold, data will not be retrieved for that date/image. The default is 1, meaning all data is retrieved
+        
+ 
+    Returns
+    -------
+    Nothing:
+            output netCDF is saved in the specified directory.
+            
+    Python dependencies required: earthengine-api, geopandas, pandas, netCDF4, xarray
+    
+    To test this function please run the following code inside a python shell after importing this module, testfile is included.
+    
+    s2_lai(geofile="./satellitetools/test.geojson", outdir="./out/", start="2019-01-01", end="2019-12-31", qi_threshold=1)       
+          
+    """
+
+    # read in the input file containing coordinates
+    df = gpd.read_file(geofile)
+
+    request = gee.S2RequestParams(start, end)
+
+    # filter area of interest from the coordinates in the input file
+    area = gee.AOI(df[df.columns[0]].iloc[0], df[df.columns[1]].iloc[0])
+
+    # calcuate qi attribute for the AOI
+    gee.ee_get_s2_quality_info(area, request)
+
+    # get the final data
+    gee.ee_get_s2_data(area, request, qi_threshold=qi_threshold)
+
+    # convert dataframe to an xarray dataset, used later for converting to netCDF
+    gee.s2_data_to_xarray(area, request)
```

Comment:
up until here, this function overlaps with s2ndvi.py which seems redundant. Two options here:
1) modify this function to pick it up from where s2ndvi.py left
2) delete/modify s2ndvi.py in favor of this one

(2) is fine because there is not much use for NDVI only, and as I said, I don't see these functions as the main functions of the module yet (see the post on our slack channel). After this PR we will have one function that gets LAI for a ROI which is already good. 

But going with (1) would be a good exercise for you to figure out how to divide the pipeline into subtasks. It could give you some design ideas. Then, for example, you would need a "master" function that calls `s2ndvi.py` first (you may also consider renaming it to something more general like `gee2bands.py`, but `s2ndvi.py` is also fine for now) and then `bands2lai.py` which calculates LAI

Reply:
(1) is exactly what I have planned, I didn't do it yet because I cannot decide for sure till we discuss the automation/db part.

Reply:
For now you can ignore the DB part, as being discussed on the slack channel, it will require some more thought. This PR can focus on providing users some stand-alone mini-workflow that calculates LAI for a ROI

Reply:
Right now if I were to follow (1), then I would remove `s2lai.py` completely, and then until we have the module ready, the LAI function will have to be used through `s2ndvi.py`(I can change its name)

Reply:
Why would you need to remove it completely? I was suggesting `s2ndvi.py` (could be renamed) would write the area data (`area.data.to_netcdf`), and `s2lai.py` (could be renamed) will take it from there, read the data back in and calculate LAI. Would that work?

Reply:
What I am suggesting is, 

1. Temporarily until the module is ready, there would be one main script to access the contents of satellitetools (`biophys_xarray.py` and `geeapi.py`.)

2. Since both `s2ndvi.py` and `s2lai.py` themselves do not perform any calculations and instead rely on `geeapi.py` and `biophys_xarray.py`, I could remove one of them. For example, `s2lai.py` uses `run_snap_biophys()` to calculate lai, I can directly call this function from `run_snap_biophys.py` instead of calling it through and intermediate script.

Reply:
> Why would you need to remove it completely? I was suggesting `s2ndvi.py` (could be renamed) would write the area data (`area.data.to_netcdf`), and `s2lai.py` (could be renamed) will take it from there, read the data back in and calculate LAI. Would that work?

This is what I am suggesting too, except that there is no need of `s2lai.py` I can directly call its contents from `biophys_xarray.py`.

Reply:
Sure, that is more in-line with (2), and I'm fine with it. Granted, `s2lai.py` would only read data back, call `bio.run_snap_biophys` and write the results. Probably not worth having it as a separate function, as I said it was only going to be an exercise.

Reply:
Since I am removing some files, should I close this PR and a open a new one? or continue here itself?

Reply:
Hmm good question, if this was my PR I would just keep working on it since the added/deleted files are small and related to the code (as opposed to the netcdf and excel data files in the previous PR). What's the procedure we want to stick here @robkooper ?

Reply:
Yeah, this is a small diff on text files, so it doesn't need a new PR